nvgpu/__init__.py

```diff
@@ -5,7 +5,8 @@
 
 def gpu_info():
     gpus = [line for line in _run_cmd(['nvidia-smi', '-L']) if line]
-    gpu_infos = [re.match('GPU ([0-9]+): ([^(]+) \(UUID: ([^)]+)\)', gpu).groups() for gpu in gpus]
+    gpu_infos = [re.match('GPU ([0-9]+): ([^(]+) \(UUID: ([^)]+)\)', gpu) for gpu in gpus]
+    gpu_infos = [info.groups() for info in gpu_infos if info is not None]
```

Comment:
In case some group is None then the values will go out of sync with the keys. Except for the case when only one or more trailing groups are None. Then the zip will be trimmed to the shorter list.

IMHO a safer way would be to filter None's after zipping with the keys.

```
gpu_infos = {key: info.groups() for key, info in zip(['index', 'type', 'uuid'], gpu_infos) if info is not None}
```

Reply:
I think I do not really understand your suggestion. The list `gpu_infos` consists of regex matches, one for each line. The match either contains three groups (since we use '+' for the group regex) or is None if there was no match. If we now filter out all `None` matches we end up with a list of matches that all have three groups. For each match we now zip the three groups to their keys `['index', 'type', 'uuid']`.
In your suggestion we end up with a dict that can only have a length of 3 at maximum and would look like the following:

`{'index': ('0', 'A100-SXM4-40GB', 'GPU-2ac434d3-1aca-57a0-597d-1d2effdba9f3'), 'type': ('1', 'A100-SXM4-40GB', 'GPU-cd9ee947-34d2-d0fe-93e0-2195b3f2fd52'), 'uuid': ('2', 'A100-SXM4-40GB', 'GPU-67051dc2-cce1-4acf-d254-e76d938eda30')}`

Reply:
Aah, sorry. I likely misread the code. Yeah, we first make matches, then skip empty matches and then take the matches groups which will be there either all or none. So there's no problem with that.